includes/compat/trait-wc-stripe-deposits.php

```diff
@@ -0,0 +1,457 @@
+	 * Whether the current cart contains a deposit.
+	 *
+	 * @since x.x.x
+	 *
+	 * @param  int $order_id
+	 * @return bool
+	 */
+	public function cart_contains_deposit() {
+		if ( ! $this->is_deposits_enabled() || ! class_exists( 'WC_Deposits_Cart_Manager' ) ) {
+			return false;
+		}
+		$cart_manager = WC_Deposits_Cart_Manager::get_instance();
+		return $cart_manager->has_deposit();
+	}
+
+	/**
+	 * Whether the current order contains a deposit.
+	 *
+	 * @since x.x.x
+	 *
+	 * @param int|\WC_Order $order_id The order ID or order object.
+	 * @return bool True if the order includes a deposit item, false otherwise.
+	 */
+	public function order_contains_deposit( $order_id ) {
+		if ( ! $this->is_deposits_enabled() || ! class_exists( 'WC_Deposits_Order_Manager' ) ) {
+			return false;
+		}
+
+		$order = wc_get_order( $order_id );
+		if ( ! $order ) {
+			return false;
+		}
+
+		$order_manager = WC_Deposits_Order_Manager::get_instance();
+		return $order_manager->has_deposit( $order );
+	}
+
+	/**
+	 * Return the top-most order function in a hierarchy.
+	 *
+	 * Payment tokens are stored against the top-most order in a hierarchy. This function
+	 * traverses the order hierarchy to find one that does not have a parent.
+	 *
+	 * Not having a parent may be because the order parent is set to zero or because
+	 * the order parent references an order that does not exist.
+	 *
+	 * @since x.x.x
+	 *
+	 * @param \WC_Order|int $order The order.
+	 * @return \WC_Order|false The top-most order in the hierarchy. False if the order does not exist.
+	 */
+	public function get_top_most_order( $order ) {
+		if ( ! $this->is_deposits_enabled() || ! class_exists( 'WC_Deposits_Order_Manager' ) ) {
+			return false;
+		}
+
+		$order = wc_get_order( $order );
+		if ( ! $order ) {
+			return false;
+		}
+
+		$order_manager = WC_Deposits_Order_Manager::get_instance();
+		return $order_manager->get_top_most_order( $order );
+	}
+
+	/**
+	 * Process a payment and store a reusable token against the order.
+	 *
+	 * @param WC_Order $order Order object.
+	 * @param bool     $retry Whether this is a retry.
+	 * @param mixed    $error Previous error.
+	 */
+	public function process_order_saving_order_token( $order_id, $retry = true, $error = false ) {
+		try {
+			$order = wc_get_order( $order_id );
+			if ( ! $order ) {
+				return false;
+			}
+
+			$future_payments = function ( $request ) {
+				$request['setup_future_usage'] = 'off_session';
+				return $request;
+			};
+
+			add_filter( 'wc_stripe_generate_create_intent_request', $future_payments );
+
+			$source   = $this->prepare_order_source( $order ); // @phpstan-ignore-line (prepare_order_source is defined in the classes that use this trait)
+			$response = $this->create_and_confirm_intent_for_off_session( $order, $source ); // @phpstan-ignore-line (create_and_confirm_intent_for_off_session is defined in the classes that use this trait)
+
+			add_filter( 'wc_stripe_generate_create_intent_request', $future_payments );
```

Comment:
This filter has already been added a few lines above. Did you mean to remove it?